### DISCUSSION TITLE
The Green Software Practitioner in 2024?

### DISCUSSION THREAD

**Discussion by russelltrow** [Green-Software-Foundation/learn#213] [General]
As we move into the new year this is a great opportunity for us to start discussing what we would like to see from the [Green Software Practitioner course](https://learn.greensoftware.foundation/) in 2024

## Planned

There are a few initiatives that we have previously discussed and are planned for this year:

* https://github.com/Green-Software-Foundation/writers/issues/185
* Translations - [Dutch](https://github.com/Green-Software-Foundation/learn/issues/195), [Chinese](https://github.com/Green-Software-Foundation/learn/issues/184), [Japanese](https://github.com/Green-Software-Foundation/learn/issues/184), [German](https://github.com/Green-Software-Foundation/learn/issues/173)
* v2.0 of the course updated based on [learning feedback with more far visuals and videos](https://github.com/Green-Software-Foundation/learn/issues/199)
* [Creation of an eLearning/SCORM course from the existing content](https://github.com/Green-Software-Foundation/learn/discussions/178)

There are also a few larger Executive Programs that will support the GSP course more broadly:

* [GSF Learn Platform, Bring the GSP course in-house](https://github.com/Green-Software-Foundation/learn-platform/issues/1) 
* [Redesign the learn.greensoftware.foundation website](https://github.com/Green-Software-Foundation/learn/issues/179)
* [Distribution of the course to other training platforms such as coursea, Pluralsight](https://github.com/Green-Software-Foundation/learn/issues/144)

As an extension to the course we have also discussed the creation of a [Certified Green Cloud Engineer Exam](https://github.com/Green-Software-Foundation/community-wg/issues/92)

## Potential

We would now love to hear from you. What do you need from the Green Software Practitioner course? How would you like to see it evolve? How else could our training support you?

* How might we answer "What next?" after completing the course - Now that I have completed the course how can I apply the knowledge and continue my journey?
* How might we have less developer-centric content? What about Design, Product and other related functions?
* How might we connect with the patterns?
* How might we provide the knowledge in alternative formats such as a [Chat bot](https://chatthing.ai/bots/dd4217cc-9b17-4ae2-8a55-d60168dff7e5/) or microlearning?

**Comment by russelltrow** [comment]
In the [Principles & Patterns project call on 9th January](https://github.com/Green-Software-Foundation/patterns/issues/293) the following ideas were raised for evolving the Green Software Practitioner and Patterns projects:

### learn.greensoftware.foundation website

* Organise the website around the concept of "Measure > Learn > Reduce"
* Use the website to bring together the best of the GSF projects and demonstrate how they connect in the real-world

### Green Software Practitioner course

* Course revamp - There is a lot of new content that we could add
* Use more videos & visuals to make the course content easier to digest
* Create Different levels of practitioner courses [similar to AWS](https://aws.amazon.com/certification/)
* Make it less developer centric
* Review whether there is too much content in the current course - does everyone really need to know everything?
* Make it easier to apply the knowledge gained - Answer "what next?" after completing the course as we are currently missing what this means for me as a practitioner

### Green Software Patterns

* Make the Patterns more accessible by linking them the core green software principles and real-world Use Cases
* Gather SCI Use Cases and link to the Patterns & Principles

### Certification

- Re-assess the proposed “Green Cloud Engineer Certification”

---

I will create a comment below for each of these issues so we can continue evolving them.

**Comment by aecurrie** [comment]
I'm going to step in with a pre-measurement suggestion. The biggest low
hanging fruit doesn't need us to measure anything other than your hosting
bill. So there are measure learn reduce loops that happen with proxy
measurements before carbon comes into it and those are probably going to be
your biggest wins

**Comment by russelltrow** [comment]
**[learn.greensoftware.foundation]**

### Idea: Organise training around the concept of "Measure > Learn > Reduce"

![alt](https://www.datocms-assets.com/56002/1670929784-screen-shot-2022-12-09-at-10-38-26-am.png?auto=compress%2Cformat&w=1920)

https://greensoftware.foundation/articles/green-software-foundation-expands-efforts-to-incentivize-decarbonization-in-t

**Comment by russelltrow** [comment]
(learn.greensoftware.foundation)

### Idea: Use the website to bring together the best of the GSF projects and demonstrate how they connect in the real-world

* Lower the barrier to entry for practitioners
* A "Green Software Academy"
* Bring together Standards, tooling and best practices for creating and building green software” (all our projects in one place)
* Bring together all our knowledge projects (Patterns, Maturity, OUSE, SOGS etc..)
* Speaks to Engineering, Design, Product etc..

**Reply by russelltrow** [reply]
Discussed and developed the idea in [P&P call on 16th Jan](https://github.com/Green-Software-Foundation/patterns/issues/298)

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Course revamp - There is a lot of new content that we could add

* If we were to revamp the course what was that mean for existing practitioners?
  * Would it be confusing for those who have already gained the certification?
* Are we better building on top of the the course rather than amending the existing content?
* Others are already developing new content so should we be referencing others work rather than developing more for ourselves?
  * Should we be duplicating effort?
  * e.g. https://www.greenskills.inco-group.co/
  * e.g. https://www.unssc.org/courses/digital4sustainability-learning-path
  * e.g. https://lfca.earth/event/digital-sustainability/
* Rather than a revamp, what is 2.0? - What is the increment we want to see?
* How can we tackle any potential gaps in the content around specialisation (non-developer)
* Received feedback that lots of the examples are too trivial - Missing the real-world application
* Could well be better to connect to our tooling and use-cases

A revamp feels a lower priority

**Reply by russelltrow** [reply]
Discussed and developed the idea in [P&P call on 16th Jan](https://github.com/Green-Software-Foundation/patterns/issues/298)

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Use more videos & visuals to make the course content easier to digest

* https://github.com/Green-Software-Foundation/learn/issues/199

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Create Different levels of practitioner courses similar to AWS

- Create Different levels of practitioner courses [Similar to AWS](https://aws.amazon.com/certification/)
- Connect different levels to Maturity Matrix

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Make it less developer centric

* Questioned whether the course truly is developer-centric or it just appears that way
  * Does it not really speak to non-developers rather than actually being developer-centric
  * Is it just a positioning thing?
  * Is it a language thing? Are we speaking in a language that is familiar to non-developers?
* We should assess which content is applicable to everyone vs developers
* Having clear next steps from the course be more powerful than trying to cater to everyone in the one course
* Connects to [Idea: Review whether there is too much content in the current course](https://github.com/Green-Software-Foundation/learn/discussions/213#discussioncomment-8145229)

**Reply by russelltrow** [reply]
Discussed and developed the idea in [P&P call on 16th Jan](https://github.com/Green-Software-Foundation/patterns/issues/298)

**Reply by cf-almeida** [reply]
An important point would be to put some thought on the Principles as well. Efficiency is a core and mostly the only aspect we are focused on. That makes GSF very engineering centered and may reduce the reach to other audiences. Even if the scope stays within efficiency, it is interesting to maybe explore what efficiency could mean to designers and product managers beyond energetic efficiency. For example, how is efficiency seen in a context of user experience? How can design friction and the absence of it impact on user behavior and therefore on the gains created by engineering efficiency? How can that impact as well on cost, price fluctuations, profit margins and potential creation of business models that are supportive also a sustainable and fair future instead of leading to even greater levels of exploitation?

**Reply by Mettenim** [reply]
Would also like to add that there are gains to be made in the area of service design as well. This often involves how multiple software tools work with each other in tandem within an ecosystem, and ways to ease the end-to-end processes (both technical and user facing)

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Review whether there is too much content in the current course

* Does everyone really need to know everything?
* Split into different levels of depth depending how deep you want to go
* Content is a lot more complex than first appears
* Some elements are too specific

**Comment by russelltrow** [comment]
(Green Software Practitioner)

### Idea: Make it easier to apply the knowledge gained

Answer "what next?" after completing the course as we are currently missing what this means for me as a practitioner

**Comment by russelltrow** [comment]
(Green Software Patterns)

### Idea: Make the Patterns more accessible by linking them the core green software principles and real-world Use Cases

* Make more accessible by connecting to the Principles
* Link each Pattern back to the 3 core principles
  * Carbon Efficiency: Emit the least amount of carbon possible.
  * Energy Efficiency: Use the least amount of energy possible.
  * Carbon Awareness: Do more when the electricity is cleaner and do less when the electricity is dirtier.
* Connect to a use case - “Here is how it might look”
* Use Cases that are potentially not directly connected to SCI
* Potentially other measurements such as Cost, "How a person's work impacts green software" > Overall productivity
* CAST are developing case studies based on https://www.omg.org/spec/ASCRSM/1.0/Beta1/About-ASCRSM#document-metadata to allow a direct link between the project and reduction - Plan to publish a selection e.g. Italian Highways - Expect to have results this year

Agreed that Use Cases are a very powerful way prove the value of our projects

**Reply by russelltrow** [reply]
Discussed and developed the idea in [P&P call on 16th Jan](https://github.com/Green-Software-Foundation/patterns/issues/298)

**Comment by russelltrow** [comment]
(Green Software Patterns)

### Idea: Gather SCI Use Cases and link to the Patterns & Principles

* e.g. https://github.com/Green-Software-Foundation/sci-guide/issues/31
* Prove that you can reduce your carbon footprint by applying these principles & patterns
  * Having SCI gives me proof and allows me to demonstrate the impact 
* It can be a challenge to implement SCI so a higher barrier to entry
* One of the biggest challenges is simply having the data
* Use Cases are particularly interesting as have the blessing of companies - Significant effort to get them published
* We need to speed up the process of creating Use Cases
* Tammy is applying SCI to her dog food business - A great example of SCI application outside the norm
* https://sci-guide.greensoftware.foundation/CaseStudies
  * Focus on the DOW/MSFT one first - It's almost there

**Reply by russelltrow** [reply]
Discussed and developed the idea in [P&P call on 16th Jan](https://github.com/Green-Software-Foundation/patterns/issues/298)

**Comment by russelltrow** [comment]
(Certification)

### Idea: “Green Cloud Engineer Certification”

* https://github.com/Green-Software-Foundation/community-wg/issues/92
